cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -291,6 +295,47 @@ public void checkGoals(CruiseControlConfiguration configuration) {
         }
     }
 
+    /**
+     * @return  True if API users are enabled. False otherwise.
+     */
+    public boolean apiUsersEnabled() {
+        return apiUsersEnabled;
+    }
+
+    /**
+     * Checks if Cruise Control spec has valid ApiUsers config.
+     *
+     * @param ccSpec The Cruise Control spec to check.
+     */
+    private static boolean isApiUsersConfigEnabled(CruiseControlSpec ccSpec)  {
+        ApiUsers apiUsers = ccSpec.getApiUsers();
+        if (apiUsers != null)    {
+            if (apiUsers.getValueFrom() == null
+                    || apiUsers.getValueFrom().getSecretKeyRef() == null
+                    || apiUsers.getValueFrom().getSecretKeyRef().getName() == null
+                    || apiUsers.getValueFrom().getSecretKeyRef().getKey() == null) {
+                throw new InvalidResourceException("Resource requests custom ApiUsers config but doesn't specify the secret name and key");
```

Comment:
I'm not sure if anyone else would understand what this means. Maybe you could say something like `The configuring of Cruise Control REST API users in spec.cruiseControl.apiUsers is invalid.`.

Also, aren't the users always enabled and the only thing that changes is whether we add user-defined users as well or not?

Reply:
> Also, aren't the users always enabled and the only thing that changes is whether we add user-defined users as well or not?

Yes, just updated the method names to hopefully make this more clear